jcommune-view/jcommune-web-controller/src/test/java/org/jtalks/jcommune/web/controller/PostControllerTest.java

```diff
@@ -223,13 +224,13 @@ public void testSubmitAnswerValidationFail() throws NotFoundException {
         BeanPropertyBindingResult resultWithErrors = mock(BeanPropertyBindingResult.class);
         when(resultWithErrors.hasErrors()).thenReturn(true);
         //invoke the object under test
-        ModelAndView mav = controller.create(null, TOPIC_ID, getDto(), resultWithErrors);
+        ModelAndView mav = controller.create(null, TOPIC_ID, getDto(), resultWithErrors, new RedirectAttributesModelMap());
 
         //check expectations
         verify(topicModificationService, never()).replyToTopic(anyLong(), anyString(), eq(BRANCH_ID));
 
         //check result
-        assertViewName(mav, "topic/postList");
+        assertEquals(mav.getViewName(), "redirect:/topics/error/" + TOPIC_ID + "?page=null");
```

Comment:
@RuslanSh , before fix the test was check assertion about redirection to the topic view page. Now this one check another behaviour - redirection when validation result had errors.
Perhap it would be better to check these two distinct behaviours in two separated test-methods?